src/vs/code/electron-main/main.ts

```diff
@@ -223,25 +229,29 @@ class CodeMain {
 		return instanceEnvironment;
 	}
 
-	private initServices(environmentMainService: IEnvironmentMainService, policyService: IPolicyService, configurationService: ConfigurationService, stateMainService: StateMainService): Promise<unknown> {
+	private async initServices(environmentMainService: IEnvironmentMainService, userDataProfilesMainService: UserDataProfilesMainService, configurationService: ConfigurationService, stateMainService: StateMainService): Promise<unknown> {
+		// State service
+		await stateMainService.init();
+
 		return Promises.settled<unknown>([
 
 			// Environment service (paths)
 			Promise.all<string | undefined>([
 				environmentMainService.extensionsPath,
 				environmentMainService.codeCachePath,
 				environmentMainService.logsPath,
-				environmentMainService.globalStorageHome.fsPath,
+				userDataProfilesMainService.defaultProfile.globalStorageHome.fsPath,
 				environmentMainService.workspaceStorageHome.fsPath,
 				environmentMainService.localHistoryHome.fsPath,
 				environmentMainService.backupHome
 			].map(path => path ? FSPromises.mkdir(path, { recursive: true }) : undefined)),
 
-			// Configuration service
-			configurationService.initialize(),
+			// User Data Profiles Service
+			userDataProfilesMainService.init(stateMainService)
```

Comment:
With the introduction of `async`, there should be really no need to use `then` or any related `Promise` function anymore. Using `then` makes the code really hard to read and understand.